client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/TransportFactory.java

```diff
@@ -67,4 +67,6 @@
    void reset(byte[] cacheName);
 
    boolean trySwitchCluster(byte[] cacheName);
+
+   boolean trySwitchCluster(String clusterName);
```

Comment:
I think you need a new name, there's already a `trySwitchCluster(byte[] cacheName)`, which I presume handles the automatic switch.